File: 509. Fibonacci Number.go
func fib(n int) int {
	if n == 0 {
		return 0
	}
	if n == 1 {
		return 1
	}

	first := 0
	second := 1
	sum_ := 0
	for i := 2; i <= n; i++ {
		sum_ = first + second
		first = second
		second = sum_
	}
	return sum_
}

File: 509. Fibonacci Number.cpp
class Solution
{
public:
    int _fib(int N, vector<int> &nums)
    {
        if (nums[N] >= 0)
            return nums[N];

        nums[N] = _fib(N - 2, nums) + _fib(N - 1, nums);
        return nums[N];
    }

    int fib(int N)
    {
        if (N <= 1)
            return N;

        vector<int> nums(N + 1, -1);
        nums[0] = 0;
        nums[1] = 1;
        return _fib(N - 2, nums) + _fib(N - 1, nums);
    }
};